File: data/transcoder_evaluation_gfg/go/NTH_NON_FIBONACCI_NUMBER.go
package main

import (
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(n int) int {
	var (
		prevPrev int = 1
		prev     int = 2
		curr     int = 3
	)
	for n > 0 {
		prevPrev = prev
		prev = curr
		curr = prevPrev + prev
		n = n - (curr - prev - 1)
	}
	n = n + (curr - prev - 1)
	return prev + n
}
//TOFILL
func main() {
	var (
		n_success int     = 0
		param0    []int = []int{76, 91, 62, 65, 83, 57, 76, 6, 2, 86}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(param0[i]) == f_gold(param0[i]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}

File: data/transcoder_evaluation_gfg/rust/NTH_NON_FIBONACCI_NUMBER.rs
#![feature(main)]
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#[no_mangle]
pub unsafe fn abs(mut x: i32)
 -> i32 {
    return x.abs();
}

#[no_mangle]
pub unsafe fn min(mut x: i32, mut y: i32)
 -> i32 {
    return if x < y { x } else { y };
}
#[no_mangle]
pub unsafe fn max(mut x: i32, mut y: i32)
 -> i32 {
    return if x > y { x } else { y };
}

pub unsafe fn f_gold(mut n: i32) -> i32 {
    let mut prevPrev: i32 = 1 as i32;
    let mut prev: i32 = 2 as i32;
    let mut curr: i32 = 3 as i32;
    while n > 0 as i32 {
        prevPrev = prev;
        prev = curr;
        curr = prevPrev + prev;
        n = n - (curr - prev - 1 as i32)
    }
    n = n + (curr - prev - 1 as i32);
    return prev + n;
}
//TOFILL
unsafe fn main_0() -> i32 {
    let mut n_success: i32 = 0 as i32;
    let mut param0: [i32; 10] =
        [76 as i32, 91 as i32, 62 as i32,
         65 as i32, 83 as i32, 57 as i32,
         76 as i32, 6 as i32, 2 as i32,
         86 as i32];
    let mut i: i32 = 0 as i32;
    while i < param0.len() as i32 {
        if f_filled(param0[i as usize]) == f_gold(param0[i as usize]) {
            n_success += 1 as i32
        }
        i += 1
    }
    println!("{} {} {} {} {}", "#Results:", " ", n_success, ", ", param0.len() as i32);
    return 0 as i32;
}
#[main]
pub fn main() { unsafe { ::std::process::exit(main_0() as i32) } }
